podcastfy/core/podcast.py

```diff
@@ -0,0 +1,361 @@
+from podcastfy.core.transcript import TranscriptSegment, Transcript
+from podcastfy.core.tts_configs import TTSConfig
+
+
+class PodcastState(Enum):
+    """Enum representing the different states of a podcast during creation."""
+    INITIALIZED = 0  # Initial state when the Podcast object is created
+    TRANSCRIPT_BUILT = 1  # State after the transcript has been generated
+    AUDIO_SEGMENTS_BUILT = 2  # State after individual audio segments have been created
+    STITCHED = 3  # Final state after all audio segments have been combined
+
+
+def podcast_stage(func):
+    """Decorator to manage podcast stage transitions."""
+
+    @wraps(func)
+    def probably_same_func(method, func):
+        return method.__func__.__name__ == func.__name__
+
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        current_method = self._next_stage_methods[self.state]
+        print(f"Current state: {self.state.name}")
+        print(f"Executing: {func.__name__}")
+        if not probably_same_func(current_method, func) and not self._reworking:
+            print(f"Cannot execute {func.__name__} in current state {self.state.name}. Skipping.")
+            raise Exception(f"Cannot execute {func.__name__} in current state {self.state.name}")
+
+        try:
+            result = func(self, *args, **kwargs)
+            next_state = PodcastState(self.state.value + 1)
+            self.state = next_state or self.state
+            print(f"Done!")
+            return result
+        except Exception as e:
+            print(f"Error in {func.__name__}: {str(e)}")
+            raise
+
+    return wrapper
+
+
+class Podcast:
+    """Main class for podcast creation and management."""
+
+    def __init__(self, content: List[Content], llm_backend: LLMBackend,
+                 audio_manager: AudioManager,
+                 characters: Optional[List[Character]] = None):
+        """
+        Initialize a new Podcast instance.
+
+        Args:
+            content (str): The raw content to be processed into a podcast.
+            llm_backend (LLMBackend): The language model backend for generating the transcript.
+            tts_backends (List[TTSBackend]): List of available TTS backends.
+            audio_temp_dir (Optional[str]): Path to a temporary directory for audio files. If None, a temporary
+                directory will be created.
+            characters (List[Character]): List of characters participating in the podcast.
+            default_tts_n_jobs (int, optional): The default number of concurrent jobs for TTS processing.
+                Defaults to 1.
+
+        Raises:
+            ValueError: If a character's preferred TTS backend is not available.
+        """
+        self.content = content
+        self.llm_backend = llm_backend
+        self.characters: Dict[str, Character] = {char.name: char for char in (characters or [Character("Host", "Podcast host", {}), Character("Guest", "Expert guest", {})])}
```

Comment:
should take from conversation_config (default or user provided) instead of static values